packages/simple-validators/src/validators.ts

```diff
@@ -207,22 +206,122 @@ export function validateEnum<T>(
+ * Parses `null` dates as "now".
  */
-export function validateDate(input: any, opts: ValidationOptions) {
-  const date = new Date(input);
-  if (!date.getDate()) {
+export function validateDate(input: any, opts: ValidationOptions & { dateIsUTC?: boolean }) {
+  // String format dates
+  if (typeof input === 'string') {
+    // Try ISO 8601
+    let match = input.match(ISO8601_DATE_PATTERN);
+    if (match) {
+      const [year, month, day, tail] = match.slice(1, 5);
+      // Is a timestamp component present?
+      if (tail !== undefined) {
+        validationWarning(
+          `invalid date "${input}" - ISO 8601 dates must not include a time component ("${tail}"), which has been ignored`,
+          opts,
+        );
+      }
+      const fullMonth = parseInt(month);
+      const fullDay = parseInt(day);
+      if (fullMonth < 1 || 12 < fullMonth || fullDay < 1 || 31 < fullDay) {
+        return validationError(
+          `invalid date "${input}" - must be ISO 8601 or (specialised) RFC 2822 format calendar date`,
+          opts,
+        );
+      }
+      return [year, month, day].filter((item) => item).join('-');
+    }
+
+    // Try a variant of RFC2822
+    match = input.match(RFC2822_DATE_PATTERN);
+    if (match) {
+      const [day, month, year, tail] = match.slice(2, 6);
+
+      // Is a timestamp component present?
+      if (tail !== undefined) {
+        validationWarning(
+          `invalid date "${input}" - specialised RFC 2822 dates must not include a time component ("${tail}"), which has been ignored`,
+          opts,
+        );
+      }
+
+      const numericYear = parseInt(year);
+
+      // Convert Jan to 1 etc.
+      const numericMonth = MONTH_TO_NUMBER.get(month)!;
+
+      // Validate numeric day, which cannot be 0, 01, and cannot exceed 31
+      const numericDay = parseInt(day);
+      if (day.startsWith('0') || numericDay < 1 || 31 < numericDay) {
+        return validationError(
+          `invalid date "${input}" - must be ISO 8601 or (specialised) RFC 2822 format calendar date`,
+          opts,
+        );
+      }
+      return buildISO8601DateString(numericYear, numericMonth, numericDay);
+    }
+    // Try falling back on JS parsing and assume it's parsed in the local timezone
+    const parsed = Date.parse(input);
+    if (isNaN(parsed)) {
+      return validationError(
+        `invalid date "${input}" - must be ISO 8601 or (specialised) RFC 2822 format calendar date`,
+        opts,
+      );
+    }
+    const localDate = new Date(parsed);
+    const result = buildISO8601DateString(
+      localDate.getFullYear(),
+      localDate.getMonth() + 1,
+      localDate.getDate(),
+    );
+    validationWarning(
+      `invalid date "${input}" - must be ISO 8601 or (specialised) RFC 2822 format calendar date. Attempting to interpret date as "${result}"`,
+      opts,
+    );
+    return result;
+  }
+  // Handle null-like as "current time"
+  else if (input == null) {
```

Comment:
Can we add some docs - and update the frontmatter section with the new recommendation? I think this should just be removing quite a bit?